### PR TITLE
OSMT-300: Show categories with totals on collection detail pages.

### PR DESCRIPTION
**Pull request by trs-aaron**
resolves https://github.com/wgu-opensource/osmt/issues/300

- Added KeywordCount to store a Keyword and total count.
- Added Map of all KeywordTypes to KeywordCounts that include all keywords and number of Skills with Keyword to Collection.
- Added Pill & PillGroup components to display Categories on Collection pages.
- Added Categories with counts to Collection public & manage pages.
- On Collection manage pages clicking on Category pill filters skills based on Category.

### CODE REVIEW

**Comment by ottonomy** [comment]
@JohnKallies I'm seeing lots of ❌ across builds today https://github.com/wgu-opensource/osmt/pull/363 with the same error as the build from this PR. @trs-aaron indicates that the tests ran successfully on local, and given that this same error is occurring across other branches, it looks like it is not related to the commits in this PR.

**Comment by JohnKallies** [comment]
@ottonomy @trs-aaron yes, we are having issues with our CI build in GitHub runners. We have been looking at this for several days now, and aren't getting very far (if you have any ideas, we'd love to hear them).

We acknowledge that the failure is not related to @trs-aaron's changes.

**Comment by trs-aaron** [comment]
collection-categories_manage_light
collection-categories_public_dark
collection-categories_public_light
collection-categories_manage_dark

**Comment by ottonomy** [comment]
@JohnKallies The recent pr #262 caused some drift and at least a minor conflict based on the order that these PRs were reviewed and merged. Can your team cover merging your changes into this branch and resolving conflicts as part of your review?

**Review by manuel-delvillar — CHANGES_REQUESTED** [review]
We can have a collection where all RSDs are without categories, so in this case we can hide "categories"

Screenshot 2023-04-21 at 11 06 52

<br>
If we have a collection with a lot of categories, I think we can have an expansion panel (collapse) or a "Show all / Show more" if you really want to see all pills.

chrome-capture-2023-3-21 (1)

File: ui/src/app/core/pill/pill.component.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  get primaryLabel(): string {
+    return (this.control) ? this.control.primaryLabel : ''
+  }
+
+  get secondaryLabel(): string | undefined {
+    return (this.control) ? this.control.secondaryLabel : undefined
+  }
+
+  onClick() {
```
Comment (manuel-delvillar) [inline ui/src/app/core/pill/pill.component.ts:33]: Probably can be `onClick(): void` to have a type def on all methods.

File: ui/src/app/core/pill/group/pill-group.component.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+export class PillGroupComponent<TValue extends AbstractPillControl>{
+
+  @Input() pillControls: TValue[] = []
+  @Output() pillClicked: EventEmitter<TValue> = new EventEmitter()
+
+  get hasPillClickedObservers(): boolean {
+    return this.pillClicked.observers.length > 0
+  }
+
+  onPillClicked(pill: TValue) {
```
Comment (manuel-delvillar) [inline ui/src/app/core/pill/group/pill-group.component.ts:17]: Probably can be `onPillClicked(): void` to have a type def on all methods.

**Review by manuel-delvillar — APPROVED** [review]
It's working, only there are comments (without required changes)

**Pull request merged**